### PR TITLE
pythonPackages.imbox: init at 0.1.0

### PR DESCRIPTION
**Pull request by liuhh666233**
###### Description of changes

https://github.com/martinrusev/imbox

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/imbox/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ lib, buildPythonPackage, fetchFromGitHub, setuptools, chardet }:
+
+buildPythonPackage rec {
+  pname = "imbox";
+  version = "0.1.0";
```
Comment (fabaff) [inline pkgs/development/python-modules/imbox/default.nix:5]: The latest upstream release is [0.9.6](https://github.com/martinrusev/imbox/releases/tag/0.9.6). Why do we start with 0.1.0 which was never tagged by upstream?

File: pkgs/development/python-modules/imbox/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "martinrusev";
+    repo = "imbox";
+    rev = "73ef2267ee6f9af927c30be68009552fc2b0bdbe";
+    sha256 = "sha256-4qMchhpkZK3cmrtRSd18PLaz/vaIE+jBIjfJnI11SWk=";
+  };
+
+  patches = [
+    ./patches/Support-to-connect-163-mail-server.patch
+    ./patches/Support-to-parse-chinese-attachment-file-name.patch
```
Comment (fabaff) [inline pkgs/development/python-modules/imbox/default.nix:17]: Please link to the upstream issue for the patches.

File: pkgs/development/python-modules/imbox/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  ];
+
+  propagatedBuildInputs = [ setuptools chardet ];
```
Comment (fabaff) [inline pkgs/development/python-modules/imbox/default.nix:20]: `setuptools` is usually not a run-time requirement. Unless the module uses `pkg_resources` or alike.

File: pkgs/development/python-modules/imbox/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  pytestFlagsArray = [ "-n" "$NIX_BUILD_CORES" ];
```
Comment (fabaff) [inline pkgs/development/python-modules/imbox/default.nix:22]: Not needed as later is the `checkPhase` disabled.

File: pkgs/development/python-modules/imbox/default.nix (hunk 1 above)
Comment (liuhh666233) [reply pkgs/development/python-modules/imbox/default.nix]: Sorry

**Review by jonringer — COMMENTED** [review]
Please follow [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) and [manual#submitting-changes-making-patches ](https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches) and squash the fix-up commits.

 `git rebase -i` is a powerful command which achieves this, I created a small video demonstrating it's use [here](https://www.youtube.com/watch?v=OR8Q-4PLalc). A more indepth text tutorial can be found [here](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)

**Pull request closed without merging**